src/linear_algebra/cross_product.lean

```diff
@@ -0,0 +1,137 @@
+rfl
+
+lemma cross_product_anticomm (v w : fin 3 → R) :
+  - (v ×₃ w) = w ×₃ v :=
+by simp [cross_product_apply, mul_comm]
+
+lemma cross_product_anticomm' (v w : fin 3 → R) :
+  v ×₃ w + w ×₃ v = 0 :=
+by rw [add_eq_zero_iff_eq_neg, cross_product_anticomm]
+
+lemma cross_product_self (v : fin 3 → R) :
+  v ×₃ v = 0 :=
+by simp [cross_product_apply, mul_comm]
+
+/-- The cross product of two vectors is perpendicular to the first vector. -/
+lemma dot_self_cross_product (v w : fin 3 → R) :
+  v ⬝ᵥ (v ×₃ w) = 0 :=
+by simp [cross_product_apply, vec3_dot_product, mul_sub, mul_assoc, mul_left_comm]
+
+/-- The cross product of two vectors is perpendicular to the second vector. -/
+lemma dot_cross_product_self (v w : fin 3 → R) :
+  w ⬝ᵥ (v ×₃ w) = 0 :=
+by rw [← cross_product_anticomm, matrix.dot_product_neg, dot_self_cross_product, neg_zero]
+
+/-- Cyclic permutations preserve the triple product. See also `triple_product_eq_det`. -/
+lemma triple_product_permutation (u v w : fin 3 → R) :
+  u ⬝ᵥ (v ×₃ w) = v ⬝ᵥ (w ×₃ u) :=
+begin
+  simp only [cross_product_apply, vec3_dot_product,
+    matrix.head_cons, matrix.cons_vec_bit0_eq_alt0, matrix.empty_append, matrix.cons_val_one,
+    matrix.cons_vec_alt0, matrix.cons_append, matrix.cons_val_zero],
+  ring,
+end
+
+/-- The triple product of `u`, `v`, and `w` is equal to the determinant of the matrix
+    with those vectors as its rows. -/
+theorem triple_product_eq_det (u v w : fin 3 → R) :
+  u ⬝ᵥ (v ×₃ w) = matrix.det ![u, v, w] :=
+begin
+  simp only [vec3_dot_product, cross_product_apply, matrix.det_fin_three,
+    matrix.head_cons, matrix.cons_vec_bit0_eq_alt0, matrix.empty_vec_alt0, matrix.cons_vec_alt0,
+    matrix.vec_head_vec_alt0, fin.fin_append_apply_zero, matrix.empty_append, matrix.cons_append,
+    matrix.cons_val', matrix.cons_val_one, matrix.cons_val_zero],
+  ring,
+end
+
+/-- The cross product satisfies the Leibniz lie property. -/
+lemma leibniz_cross (u v w : fin 3 → R) :
+  u ×₃ (v ×₃ w) = (u ×₃ v) ×₃ w + v ×₃ (u ×₃ w) :=
+begin
+  dsimp only [cross_product_apply],
+  ext i,
+  fin_cases i; norm_num; ring,
+end
+
+/-- The three-dimensional vectors together with the operations + and ×₃ form a Lie ring.
+    Note we do not make this an instance as a conflicting one already exists
+    via `lie_ring.of_associative_ring`. -/
+def cross_product.lie_ring : lie_ring (fin 3 → R) :=
+{ bracket := λ u v, u ×₃ v,
+  add_lie := linear_map.map_add₂ _,
+  lie_add := λ u, linear_map.map_add _,
+  lie_self := cross_product_self,
+  leibniz_lie := leibniz_cross,
+  ..pi.add_comm_group }
+
+local attribute [instance] cross_product.lie_ring
+
+lemma cross_cross  (u v w : fin 3 → R) :
+  (u ×₃ v) ×₃ w = u ×₃ (v ×₃ w) - v ×₃ (u ×₃ w) :=
+lie_lie u v w
+
+/-- For a cross product of three vectors, their sum over the three even permutations is equal
+    to the zero vector. -/
+theorem jacobi_identity (u v w : fin 3 → R) :
+  u ×₃ (v ×₃ w) + v ×₃ (w ×₃ u) + w ×₃ (u ×₃ v) = 0 :=
+lie_jacobi u v w
```

Comment:
One more lemma we should probably include:
```lean
theorem cross_dot_cross (u v w x : fin 3 → R) :
  (u ×₃ v) ⬝ᵥ (w ×₃ x) = (u ⬝ᵥ w) * (v ⬝ᵥ x) - (u ⬝ᵥ x) * (v ⬝ᵥ w) :=
begin
  simp only [vec3_dot_product, cross_product, cons_append, cons_vec_bit0_eq_alt0,
    cons_val_one, cons_vec_alt0, linear_map.mk₂_apply, cons_val_zero, head_cons, empty_append],
  ring_nf,
end
```
This is the identity that's special to dimension 3 (so not an identity shared by other Lie algebras, if we used the Lie bracket for the cross product and the Killing form for the dot product).